src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/CreateOrganizationOverlay.tsx

```diff
@@ -0,0 +1,209 @@
+
+// Styles
+import './CreateOrganizationOverlay.scss'
+
+// Contexts
+import {
+  CreateOrgProvider,
+  CreateOrgContext,
+  CreateOrgError,
+} from 'src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/CreateOrganizationContext'
+
+// Types
+import {RemoteDataState} from 'src/types'
+import {notify} from 'src/shared/actions/notifications'
+import {orgCreateSuccess} from 'src/shared/copy/notifications'
+import {
+  ServerError,
+  UnauthorizedError,
+  UnprocessableEntityError,
+} from 'src/types/error'
+
+const DEFAULT_ERROR_MESSAGE = 'This is a required field'
+const NAME_CONFLICT_MESSAGE =
+  'This organization name already exists. Please choose a different name.'
+const errorStyle = {
+  margin: '20px 0',
+}
+
+const errorMap = {
+  [CreateOrgError.ServerError]: 'Organization creation failed.',
+  [CreateOrgError.NameConflict]: NAME_CONFLICT_MESSAGE,
+  [CreateOrgError.Unauthorized]: 'Not authorized to create organization.',
+  [CreateOrgError.ClustersFetchError]: 'Error fetching cluster information.',
+  [CreateOrgError.GenericError]:
+    'Error creating organization. Please refresh and try again in some time.',
+}
+const CreateOrgOverlay: FC = () => {
+  const {onClose} = useContext(OverlayContext)
+  const {
+    clusters,
+    orgName,
+    changeOrgName,
+    createOrg,
+    createOrgLoadingStatus,
+    error,
+    changeCreateOrgLoadingStatus,
+    setError,
+  } = useContext(CreateOrgContext)
+  const dispatch = useDispatch()
+  const [inputError, setInputError] = useState(DEFAULT_ERROR_MESSAGE)
+  const [errorMessage, setErrorMessage] = useState(null)
+  const [showError, setShowError] = useState(false)
+
+  const handleOrgNameChange = useCallback(
+    e => {
+      changeOrgName(e.target.value)
+      setShowError(false)
+      setInputError(DEFAULT_ERROR_MESSAGE)
+    },
+    [changeOrgName]
+  )
+
+  useEffect(() => {
+    if (createOrgLoadingStatus === RemoteDataState.Error) {
+      if (error === CreateOrgError.NameConflict) {
+        setShowError(true)
+        setInputError(NAME_CONFLICT_MESSAGE)
+      } else {
+        setErrorMessage(errorMap?.[error])
+      }
+    } else {
+      setShowError(false)
+    }
+  }, [error, createOrgLoadingStatus])
+
+  const handleCreateOrg = useCallback(() => {
+    if (orgName.trim() === '') {
+      setShowError(true)
+      setInputError(DEFAULT_ERROR_MESSAGE)
+      return
+    }
+
+    createOrg()
+      .then(() => {
+        onClose()
+        dispatch(notify(orgCreateSuccess()))
+      })
+      .catch(e => {
+        if (e instanceof UnauthorizedError) {
+          setError(CreateOrgError.Unauthorized)
+        } else if (e instanceof ServerError) {
+          setError(CreateOrgError.ServerError)
+        } else if (e instanceof UnprocessableEntityError) {
+          // Even though there can be different meanings/reasons for this issue.
```

Comment:
I would modify this comment to be clear on what the 'issue' is and why we're choosing to interpret the error this way. Here's an example:

```
/*
Quartz responds with a 422 status code if the user-selected org name is a duplicate of an 
org name in the current account. It may also respond with a 422 for other reasons. 

However, the UI will interpret the 422 as an org name duplicate for purposes of user feedback, 
since a 422 generated for other reasons is very unlikely, and still means that the user cannot 
create an organization using the current input values anyway.' 
*/
```